schema_based_on_diagram.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE patients (
+  id int generated by default as identity,
+  name varchar not null,
+  date_of_birth date not null,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_histories (
+  id int generated by default as identity,
+  admitted_at timestamp NOT NULL,
+  patient_id int not null references patients(id),
+  status varchar not null,
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE invoices (
+  id int generated by default as identity,
+  total_amount decimal not null,
+  generated_at timestamp not null,
+  payed_at timestamp not null,
+  medical_history_id int not null references medical_histories(id),
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE treatments (
+  id int generated by default as identity,
+  type varchar not null,
+  name varchar not null,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE invoice_items (
+  id int generated by default as identity,
+  unit_price decimal not null,
+  quantity int not null,
+  total_price decimal not null,
+  invoice_id int not null references invoices(id),
+  treatment_id int not null references treatments(id),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE medical_histories_treaments (
+  medical_history_id int not null references medical_histories(id),
+  treatment_id int not null references treatments(id),
+  PRIMARY KEY (medical_history_id, treatment_id)
+);
```

Comment:
- [ Required ] Please consider adding indexes to all your foreign key fields, this will make searches that use these columns much faster. It is also stated in the project requirements that all foreign key fields should have an index, please see the screenshot below.

## Example of an index
```SQL
  CREATE INDEX ON invoices (medical_history_id);
```
## Project requirements 
![Screenshot from 2023-02-28 15-11-45](https://user-images.githubusercontent.com/16392046/221851330-200fc3ae-7a6c-4467-8568-d9915fb41a73.png)